### PR TITLE
[docs] api docs for missing public methods on input context, output context, and dynamic output

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation

to contribute API docs to https://github.com/dagster-io/dagster/pull/15078

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15282** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15282" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15282?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/execution/context/output.py [hunk 1]
```diff
@@ -571,6 +576,11 @@ def get_output_identifier(self) -> Sequence[str]:
 
     @public
     def get_asset_identifier(self) -> Sequence[str]:
+        """The sequence of strings making up the AssetKey for the asset being stored as an output.
+        If the asset is partitioned, the identifier contains the partition key as the final element in the
+        sequence. For example, for the asset key ``AssetKey(["foo", "bar", "baz"])``, ``get_asset_identifier`` would return
+        ``["foo", "bar", "baz"]``.
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/execution/context/output.py:582]: I think this example needs to incorporate a partition key.

**Review by smackesey — APPROVED** [review]
Great except one comment

File: python_modules/dagster/dagster/_core/execution/context/output.py [hunk 2]
```diff
@@ -571,6 +576,11 @@ def get_output_identifier(self) -> Sequence[str]:
 
     @public
     def get_asset_identifier(self) -> Sequence[str]:
+        """The sequence of strings making up the AssetKey for the asset being stored as an output.
+        If the asset is partitioned, the identifier contains the partition key as the final element in the
+        sequence. For example, for the asset key ``AssetKey(["foo", "bar", "baz"])`` materialized with
+        partition key "2023-06-01", ``get_asset_identifier`` will return ``["foo", "bar", "baz", "2023-06-01"]``.
+        """
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/execution/context/output.py:583]: nice

**Review by smackesey — APPROVED** [review]

**Pull request merged**